app/src/main/java/com/github/sdp_begreen/begreen/fragments/AdviceFragment.kt

```diff
@@ -4,42 +4,197 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.TextView
+import android.widget.ExpandableListView
 import androidx.fragment.app.Fragment
-import androidx.lifecycle.lifecycleScope
 import com.github.sdp_begreen.begreen.R
-import com.github.sdp_begreen.begreen.firebase.DB
-import kotlinx.coroutines.launch
-import org.koin.android.ext.android.inject
+import com.github.sdp_begreen.begreen.adapters.ExpandableListAdapter
+import com.github.sdp_begreen.begreen.models.Group
+import com.github.sdp_begreen.begreen.models.Item
 
 /**
  * A simple [Fragment] subclass.
+ * Use the [AdviceFragment.newInstance] factory method to
  * create an instance of this fragment.
- * @param testCallback A callback received from tests, it expects the set retrieved
- *  from the database as parameter. It is null by default, should only be non null from tests.
  */
 class AdviceFragment : Fragment() {
 
-    // Get the db instance
-    private val db by inject<DB>()
-
     // The purpose of this method is to inflate the fragment layout, initialize some views,
     // and display a random piece of advice on the screen.
+    private lateinit var expandableListView: ExpandableListView
+    private lateinit var adapter: ExpandableListAdapter
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                               savedInstanceState: Bundle?): View {
 
         val view = inflater.inflate(R.layout.fragment_advice, container, false)!!
 
-        val adviceFragmentTextView = view.findViewById<TextView>(R.id.adviceFragmentTextView)
+        // TO DO : refactor to get quotes from database
+        /*val adviceFragmentTextView = view.findViewById<TextView>(R.id.adviceFragmentTextView)
 
-        // Sets the text of the adviceFragmentTextView to a random string from the adviceSet fetched from the DB.
-        lifecycleScope.launch {
-            val advicesSet: Set<String> = db.getAdvices()
-            if (advicesSet.isNotEmpty()) {
-                adviceFragmentTextView.text = advicesSet.random()
-            }
-        }
+        // TO DO : generate a random quote per category
+        // Sets the text of the adviceFragmentTextView to a random string from the adviceList variable.
+        arguments?.also { it.getStringArrayList(QUOTES)?.also { list ->  adviceFragmentTextView.text =
+            list[Random.nextInt(list.size)]
+        } }*/
+
+        expandableListView = view.findViewById(R.id.expandable_list1)
+
+        val groups = getGroups()
+
+        adapter = ExpandableListAdapter(requireContext(), groups)
+        expandableListView.setAdapter(adapter)
 
         return view
     }
+
+    private fun getGroups(): List<Group> {
```

Comment:
For now we can keep this huge list like that, but in next sprints you should think about fetching the groups from a database.

But for now don't have to implement it.

Reply:
Totally agree! I noted in my PR comment as well under to do next.